src/main/java/edu/byu/cs/autograder/QualityAnalyzer.java

```diff
@@ -39,17 +41,17 @@ public QualityOutput runQualityChecks(File stageRepo) {
         output = output.replaceAll(stageRepo.getAbsolutePath(), "");
         output = output.replaceAll(stageRepo.getPath(), "");
 
-        QualityAnalysis analysis = parseAnalysis(output);
-        float score = getScore(analysis);
-        String results = getResults(analysis);
-        String notes = getNotes(analysis);
-        return new QualityOutput(score, results, notes);
+        parseOutput(output);
```

Comment:
This section of code seems much cleaner and more readable than before. Looking good!